inspirehep/dojson/hep/rules/bd9xx.py

```diff
@@ -236,21 +235,37 @@ def references2marc(self, key, value):
     journal_artid = get_value(reference, 'publication_info.artid')
     pubnote = build_pubnote(journal_title, journal_volume, journal_pg_start,
                             journal_pg_end, journal_artid)
+
+    a_list = ['doi:{}'.format(doi)
+              for doi in get_value(reference, 'dois', [])]
+
+    persistent_identifiers = get_value(reference, 'persistent_identifiers',
+                                       default=[])
+    for identifier in persistent_identifiers:
+        pid = get_value(identifier, 'value')
```

Comment:
If you do
```python
foos = get_value(record, 'foo')
for foo in foos:
    bar = get_value(record, 'bar')
    if some_condition_on(bar):
        ...
```
you're not getting any of the benefits of `get_value`, just the fact that is ~7x slower than straight dictionary access.

In this case, where you have to filter on the `bar`, it's better to use straight dictionary accesses:
```python
for foo in force_list(record.get('foo')):
    if some_condition_on(foo.get('bar')):
        ...
```